main.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# bubble sort
+def buble(arr)
+  n_times = arr.length - 1
+  n_times.times do |i|
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i] > arr[i + 1]
+  end
+  print arr
+end
+
+# bubble_sort_by
+def bubble_sort_by(arr)
+  n_times = arr.length - 1
+  n_times.times do |i|
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if yield(arr[i], arr[i + 1]).positive?
+  end
+  puts arr
+end
```

Comment:
- [ ] Similar to the previous method, This method is not passing the tests, if I try putting: `p bubble_sort_by(["hello", "hey", "hi"]) { |left, right| left.length - right.length }` it gives: `["hey", "hi", "hello"]` instead of `["hi", "hey", "hello"]`. **Hint:** Your logic is basically ok, but you are only sorting it one time, that is why it only sorts the "hello". You need to give as many passes as necessary until everything is sorted